### ISSUE TITLE
[ANNOUNCEMENT] SlimSocial for Facebook is reborn in Flutter

### ISSUE THREAD

**Issue by rignaneseleo** [rignaneseleo/SlimSocial-for-Facebook#159]
Dear all,

I am pleased to make a long-awaited announcement: SlimSocial for Facebook, the lightweight and customizable alternative to the official Facebook app, has been updated with new features and a better-than-ever codebase.

As many of you know, I created SlimSocial when I was just 17 years old, and had to put it on hold due to my university and work commitments. But the app has always been close to my heart, and in the last weeks I have been working tirelessly to bring it up to date with the latest technology: [Flutter](https://flutter.dev/).

With this new update, SlimSocial retains its original values of lightweight design and privacy while also introducing new features, including the ability to hide ads and a better user interface. I have completely re-coded the app, making it more modern and user-friendly than ever before.

I believe that SlimSocial still has a place in today's world, even as Facebook has become less popular and phones have become more powerful. The app is customizable and easy to use, making it the perfect alternative to the official chunky Facebook app. You can now run your own user-agent, CSS, and Javascript code.

I am confident that you will find the new codebase of SlimSocial improved and easier to read. 
I am excited to share it with you and am grateful for your continued support.

Thank you for using SlimSocial for Facebook. 💙

Best regards,
Leonardo

**Comment by rignaneseleo** [comment]
The old codebase will be available in the branch `old-java-app`

**Comment by gothmog123** [comment]
you are a rock star sir

**Comment by ballo** [comment]
Is there any way to get notifications for message requests? The official apps don't have an option for this. Any help would be appreciated

**Comment by rignaneseleo** [comment]
> Is there any way to get notifications for message requests? The official apps don't have an option for this. Any help would be appreciated

I doubt it. Facebook removed messenger from every domain on web...

**Comment by ballo** [comment]
Why would you use web for Messenger, anyway? Why not their APIs?

**Comment by rignaneseleo** [comment]
their APIs are closed AFAIK. The only API there are, are for business chats.

**Comment by HarriBuh** [comment]
I would not ever have thought of you and your app returning to the scene, especially since almost 100% of all forks and wrappers of the Facebook app have died during the last years. I hope you will stay on track. Thank you so much.

**Comment by fidelius-dc** [comment]
Awesome! I'm stunned you even made it to integrate Messenger. Only thing that bothers me with the new version is that posts take up way more screen estate than before. An option to have smaller pics or viedo-previews would be great. or can this be achieved trough a custom CSS ?

**Comment by rignaneseleo** [comment]
> Awesome! I'm stunned you even made it to integrate Messenger. Only thing that bothers me with the new version is that posts take up way more screen estate than before. An option to have smaller pics or viedo-previews would be great. or can this be achieved trough a custom CSS ?

Yes, with custom CSS and custom JS there are endless editing possibility

**Comment by Krawei** [comment]
Will the app be released on F-Droid,too?

**Comment by gothmog123** [comment]
it's already there! thanks dev!

**Comment by LoneFenris** [comment]
Are android notifications planned? (E.g. comments, messages, etc)

**Comment by rignaneseleo** [comment]
> Are android notifications planned? (E.g. comments, messages, etc)

I don't think so. Without API is really complex and RAM consuming to enable the notifications. 
If the app gets popular again, I might think about it :)

**Comment by LoneFenris** [comment]
@rignaneseleo Thanks, that's fine. 

Though I believe the way other wrappers like Frost avoid excessive resources is by using a periodic check for new notifications (e.g. once per 15, 30, or 60 minutes), rather than keeping the site loaded all the time.

**Comment by ballo** [comment]
> their APIs are closed AFAIK. The only API there are, are for business chats.

lame.

So basically Meta decides how your app behaves, and not you.

**Comment by Maythesaucebewithyou** [comment]
Legend!

**Comment by edrex** [comment]
Thanks for your work over the years creating and maintaining SlimSocial! I've reached for it in various devices when i needed to access Facebook. 🎂🚀🍺

Feedback: new apk is 20MB, , while F-Droid description still claims "less than 100kb"
- description needs update.
- assuming increased size is flutter runtime/library overhead. Anything to be done to shrink/optimize the build? https://docs.flutter.dev/perf/app-size

**Comment by Meatloaf125** [comment]
Thanks for this, the new version is really great!

**Comment by duven87** [comment]
Ciao Leonardo, 
Have you seen that they've released another all-in-one app for Facebook, Instagram, and Twitter called Nora?
Well, judging by your post, I guess you're very fond of Slimsocial, hehe, but you might still be interested in merging with Nora or collaborating :)
Messenger, for example, still doesn't work.

Cheers!

**Comment by javier906** [comment]
Cómo puedo acceder

**Comment by javier906** [comment]
Ayuda

**Comment by stylexesp-eng** [comment]
Reborn

**Comment by secretmango** [comment]
Oh I guess this is why "lightweight" means 60MB... for a webview wrapper?

I dont judge, this is FOSS, but it is not lightweight.
